dsa/AVL.java

```diff
@@ -0,0 +1,201 @@
+import java.util.Scanner;
+
+public class AVL {
+
+  public static class Node{
+    int data;
+    Node left; //left child
+    Node right; //right child
+
+    int ht=1; // height
+    int bal=0; // balance
+
+    Node(int data){
+      this.data=data;
+    }
+  }
+
+  public static void main(String[] args) {
+     Node root=null;
+
+     Scanner sc=new Scanner(System.in);
+
+     while (true) {
+       System.out.println("Your tree: ");
+       display(root);
+
+       System.out.println("Enter 1 to add a node");
+       System.out.println("Enter 2 to remove a node");
+       System.out.println("Enter 3 to end the program");
+
+       int n=sc.nextInt();
+
+       if (n==1) {
+        System.out.println("Enter the value of the node: ");
+        int val=sc.nextInt();
```

Comment:
You need to handle exceptions for Scanner

Reply:
Got it.